### PR TITLE
feat: stratification of VAFs

### PR DESCRIPTION
**Pull request by johanneskoester**
containing just todos

### CODE REVIEW

**Comment by famosab** [comment]
More things need to be solved:

- Truth VCF has to be handed to `calc_precision_recall.py`
- Correct VAF record needs to be accessed (sample name?)
- Handle cases where VAF is denoted in INFO field
- Arrays need to be defined in constructor (Classification)
- VAF stratification has to be used in report

**Comment by famosab** [comment]
We found that rtg vcfeval removes INFO and FORMAT fields.

**Review by johanneskoester — COMMENTED** [review]
Great work!

File: workflow/resources/datavzrd/precision-recall-config.yte.yaml [hunk 1]
```diff
@@ -67,47 +67,108 @@ views:
...
+                  - 1.0
+                range:
+                  - "#c0e6baff"
+                  - "#abdda5ff"
+                  - "#94d391ff"
+                  - "#7bc77dff"
+                  - "#60ba6cff"
+                  - "#46ab5eff"
+                  - "#329a51ff"
+                  - "#208943ff"
```
[the remark this answers is not in the document]
Comment (BiancaStoecker) [reply workflow/resources/datavzrd/precision-recall-config.yte.yaml:127]: Done

File: workflow/scripts/calc-precision-recall.py [hunk 2]
```diff
@@ -7,88 +7,186 @@
...
+                else:
+                    vaf = None
+                self.increment_counter(self.fn, vaf)
             elif c.cls is Class.FP:
-                self.fp += 1
+                if self.stratify_by_vaf:
+                    r = list(query.fetch(record.contig, record.start, record.stop))[0]
+                    vaf = r.info[self.vaf_field_name_query][0] if self.vaf_field_query == "INFO" else r.format[self.vaf_field_name_query][0]
+                else:
+                    vaf = None
```
[the remark this answers is not in the document]
Comment (BiancaStoecker) [reply workflow/scripts/calc-precision-recall.py]: Done

File: workflow/resources/datavzrd/precision-recall-config.yte.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (BiancaStoecker) [reply workflow/resources/datavzrd/precision-recall-config.yte.yaml]: Done

**Comment by johanneskoester** [comment]
Fantastic!

**Pull request merged**